pkg/apis/core/types.go

```diff
@@ -3853,12 +3859,16 @@ type ServiceSpec struct {
 	// +optional
 	LoadBalancerSourceRanges []string
 
-	// externalTrafficPolicy denotes if this Service desires to route external
-	// traffic to node-local or cluster-wide endpoints. "Local" preserves the
-	// client source IP and avoids a second hop for LoadBalancer and Nodeport
-	// type services, but risks potentially imbalanced traffic spreading.
-	// "Cluster" obscures the client source IP and may cause a second hop to
-	// another node, but should have good overall load-spreading.
+	// externalTrafficPolicy describes the endpoint-selection policy for traffic
```

Comment:
It's not really about "received from outside the cluster" but "received at external-facing entrypoints" ?

Reply:
but with the special case for pod and node traffic, it ends up being that externalTrafficPolicy only affects traffic which both comes from outside the cluster and ends up an external-facing entrypoing.

(The internal case is not currently parallel; see https://github.com/kubernetes/kubernetes/issues/108526#issuecomment-1086080420)

Reply:
if a pod hits a node-port on a different node, it is subject to eTP (right?) but isn't "received from outside the cluster".

Words are hard.  I personally find it easier to think of it as "external-facing inputs" and "a couple special cases".  I am open to anything that makes this clearer. :)

Reply:
... seems the node is the boundary?